compiler/src/dotty/tools/dotc/core/Denotations.scala

```diff
@@ -386,19 +386,21 @@ object Denotations {
         /** Establish a partial order "preference" order between symbols.
          *  Give preference to `sym1` over `sym2` if one of the following
          *  conditions holds, in decreasing order of weight:
-         *   1. sym1 is concrete and sym2 is abstract
-         *   2. The owner of sym1 comes before the owner of sym2 in the linearization
+         *   1. sym1 exists but sym2 doesn't
+         *   2. sym1 is concrete and sym2 is abstract
+         *   3. The owner of sym1 comes before the owner of sym2 in the linearization
          *      of the type of the prefix `pre`.
-         *   3. The access boundary of sym2 is properly contained in the access
+         *   4. The access boundary of sym2 is properly contained in the access
          *      boundary of sym1. For protected access, we count the enclosing
          *      package as access boundary.
-         *   4. sym1 a method but sym2 is not.
+         *   5. sym1 a method but sym2 is not.
          *  The aim of these criteria is to give some disambiguation on access which
          *   - does not depend on textual order or other arbitrary choices
          *   - minimizes raising of doubleDef errors
          */
         def preferSym(sym1: Symbol, sym2: Symbol) =
           sym1.eq(sym2) ||
+            (sym1.exists && !sym2.exists) ||
```

Comment:
I think you can just say `!sym2.exists`. They are different, so `sym1.exists` is redundant.